src/fides/api/service/connectors/s3_connector.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_s3 import S3Schema
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_config import QueryConfig
+from fides.api.util.aws_util import get_aws_session
+from fides.api.util.collection_util import Row
+from fides.connectors.models import ConnectorFailureException
+
+
+class S3Connector(BaseConnector):
+    """
+    AWS S3 Connector - this is currently used just to test connections to S3.
+
+    NOTE: No DSR processing is yet supported for S3.
+    """
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for s3"""
+        config = S3Schema(**self.configuration.secrets or {})
+        return get_aws_session(
```

Comment:
this will raise an exception if authentication fails - that'll be caught by the wrapping `test_connection` function and handled appropriately there, which is fine for now.